src/Universe.cpp

```diff
@@ -817,6 +660,109 @@ Universe* Universe::clone() {
   return clone;
 }
 
+
+/**
+  * @brief  Calculates the boundary locations and 
+  *         conditions (VACUUM or REFLECTIVE) at the
+  *         maximum and minimum reachable coordinates in the Universe
+  */
+void Universe::calculateBoundaries() {
+
+  /* Calculate the minimum reachable x-coordinate in the geometry and store it
+   * in _x_min */
+  std::map<int, Cell*>::iterator c_iter;
+  std::map<int, surface_halfspace*>::iterator s_iter;
+  Surface* surf;
+  int halfspace;
+
+  /* Calculate the maximum and minumum reachable coordinates in the geometry
+     and store them in class variables */
+  double min_x = -std::numeric_limits<double>::infinity();
+  double max_x = std::numeric_limits<double>::infinity();
+  double min_y = -std::numeric_limits<double>::infinity();
+  double max_y = std::numeric_limits<double>::infinity();
+  double min_z = -std::numeric_limits<double>::infinity();
+  double max_z = std::numeric_limits<double>::infinity();
+
+  boundaryType bc_x_min = BOUNDARY_NONE;
+  boundaryType bc_x_max = BOUNDARY_NONE;
+  boundaryType bc_y_min = BOUNDARY_NONE;
+  boundaryType bc_y_max = BOUNDARY_NONE;
+
+  /* Check if the universe contains cells with boundaries */
+  for (c_iter = _cells.begin(); c_iter != _cells.end(); ++c_iter) {
+    std::map<int, surface_halfspace*> surfs = c_iter->second->getSurfaces();
+
+    for (s_iter = surfs.begin(); s_iter != surfs.end(); ++s_iter) {
+      surf = s_iter->second->_surface;
+      halfspace = s_iter->second->_halfspace;
+
+      /* x min and boundarytype */
+      if (surf->getSurfaceType() == XPLANE && halfspace == +1 &&
+          surf->getBoundaryType() != BOUNDARY_NONE) {
+        min_x = surf->getMinX(halfspace);
+        bc_x_min = surf->getBoundaryType();
+      }
+
+      /* x max and boundarytype */
+      if (surf->getSurfaceType() == XPLANE && halfspace == -1 &&
+          surf->getBoundaryType() != BOUNDARY_NONE) {
+        max_x = surf->getMaxX(halfspace);
+        bc_x_max = surf->getBoundaryType();
+      }
+
+      /* y min and boundarytype */
+      if (surf->getSurfaceType() == YPLANE && halfspace == +1 &&
+          surf->getBoundaryType() != BOUNDARY_NONE) {
+        min_y = surf->getMinY(halfspace);
+        bc_y_min = surf->getBoundaryType();
+      }
+
+      /* y max and boundarytype */
+      if (surf->getSurfaceType() == YPLANE && halfspace == -1 &&
+          surf->getBoundaryType() != BOUNDARY_NONE) {
+        max_y = surf->getMaxY(halfspace);
+        bc_y_max = surf->getBoundaryType();
+      }
+    }
+  }
+
+  /* If boundaries were not found, get them from the bounding boxes of the
+   * cells. Also get z boundaries */
+  for (c_iter = _cells.begin(); c_iter != _cells.end(); ++c_iter) {
```

Comment:
After some work in ClosedMOC I found this doesn't actually work ... we need to check to make sure that the min/max haven't already been found before entering this loop